emr-serverless-sample/README.md

```diff
@@ -0,0 +1,112 @@
+
+We will run a Java Spark job on EMR Serverless using a simple Java "Hello World" example in this example.
+
+## Prerequisites
+
+* LocalStack
+* `aws` CLI & `awslocal` script
+* Docker
+* Java and Maven
+
+## Installation
+
+Before creating the EMR Serverless job, we need to create a JAR file containing the Java code. To do this, run the following command:
+
+```bash
+cd hello-world
+mvn package
+```
+
+Next, we need to create an S3 bucket to store the JAR file. To do this, run the following command:
+
+```bash
+export S3_BUCKET=test
+awslocal s3 mb s3://$S3_BUCKET
+```
+
+You can now copy the JAR file to the S3 bucket:
+
+```bash
+awslocal s3 cp target/java-demo-1.0.jar s3://${S3_BUCKET}/code/java-spark/ 
+```
+
+## Creating the EMR Serverless Job
+
+Specify the ARN for the EMR Serverless job with the following command:
+
+```bash
+export JOB_ROLE_ARN=arn:aws:iam::000000000000:role/emr-serverless-job-role
+```
+
+We can now create an EMR Serverless application, which will run Spark 3.3.0. Run the following command:
+
+```bash
+awslocal emr-serverless create-application \
+  --type SPARK \
+  --name serverless-java-demo \
+  --release-label "emr-6.9.0" \
+    --initial-capacity '{
+        "DRIVER": {
+            "workerCount": 1,
+            "workerConfiguration": {
+                "cpu": "4vCPU",
+                "memory": "16GB"
+            }
+        },
+        "EXECUTOR": {
+            "workerCount": 3,
+            "workerConfiguration": {
+                "cpu": "4vCPU",
+                "memory": "16GB"
+            }
+        }
+    }'
+```
+
+You can retrieve the Application ID from the output of the command, and export it as an environment variable:
+
+```bash
+export APPLICATION_ID=<application-id>
+```
+
+Start the EMR Serverless application:
+
+```shell
+awslocal emr-serverless start-application \
+    --application-id $APPLICATION_ID
+```
+
+## Running the EMR Serverless Job
+
+You can now run the EMR Serverless job:
+
+```bash
+awslocal emr-serverless start-job-run \
+    --application-id $APPLICATION_ID \
+    --execution-role-arn $JOB_ROLE_ARN \
+    --job-driver '{
+        "sparkSubmit": {
+            "entryPoint": "s3://'${S3_BUCKET}'/code/java-spark/java-demo-1.0.jar",
+            "sparkSubmitParameters": "--class HelloWorld"
+        }
+    }' \
+    --configuration-overrides '{
+        "monitoringConfiguration": {
+            "s3MonitoringConfiguration": {
+                "logUri": "s3://'${S3_BUCKET}'/logs/"
+            }
+        }
+    }'
+```
+
+The Spark logs will be written to the S3 bucket specified in the `logUri` parameter. You can stop the EMR Serverless application with the following command:
+
+```bash
+awslocal emr-serverless stop-application \
+    --application-id $APPLICATION_ID
+```
+
+## Note
+
+Currently, LocalStack's EMR Serverless implementation is basic and just allows you to create applications and job runs, to run your Spark/PySpark jobs locally.
```

Comment:
Not sure this note is required - creating applications and job runs is actually the core functionality of EMR Serverless (they've deliberately kept the API simple and easy to use). If we want to keep this note, I'd point out more specifically which features we're planning to add in the future (or what's currently missing), but overall I think it could be removed. 👍